src/TextInput.js

```diff
@@ -1,167 +1,149 @@
-import React, { Component } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
+const TextInput = props => {
+  const [id, setId] = useState(props.id || idgen());
 
-    this.id = props.id || idgen();
+  const [idSetable, setIdSetable] = useState(true);
 
-    if (props.password) {
-      this.id = `password${idgen()}`;
-    }
+  const inputRef = useRef(null);
+
+  // eslint-disable-next-line react/prop-types
+  const dataLength = props['data-length'];
+
+  useEffect(() => {
+    dataLength && M.CharacterCounter.init(inputRef);
+  }, [dataLength]);
 
-    if (props.email) {
-      this.id = `email${idgen()}`;
+  useEffect(() => {
+    if (typeof M !== 'undefined') {
+      M.updateTextFields();
     }
```

Comment:
as mentioned, let's remove this check as we want it to fail with reason if M is not defined :)

Reply:
I could have sworn I got that one :P

Reply:
Also I have removed the test that asserts it wont fail if M is not defined.